src/js/components/SharedComponents/CalendarRangeDatePicker.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { PropTypes } from 'react';
+import DayPicker, { DateUtils } from 'react-day-picker';
+
+import Moment from 'moment';
+import Navbar from './CalendarRangeDatePickerNavbar';
+
+const propTypes = {
+  numberOfMonths: PropTypes.number,
+  onSelect: PropTypes.func.isRequired,
+  minmaxDates: PropTypes.object
+};
+
+const defaultProps = {
+  numberOfMonths: 2,
+  minmaxDates: {
+      maxDate: '',
+      minDate: ''
+  }
+};
+
+export default class CalendarRangeDatePicker extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.dateValues = {};
```

Comment:
`this.dateValues` is redundant because it gets set to `this.state` on line 66. When it is referenced in `sendToFilters`, you can lookup `this.state.to` / `this.state.from` instead.